### PR TITLE
docs: add blog for Vault

### PR DESCRIPTION
**Pull request by yzeng25**
Changes:
Add blog for Vault integration.

### CODE REVIEW

**Comment by juzhiyuan** [comment]
cc @guoqqqi @oil-oil to have a review 😉

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+- 插件生态
+description: 本文为大家带来了 Apache APISIX 即将发布的 Vault 插件以及相关细节。在为服务提供高并发低延迟的卓越性能的同时，为服务的安全保驾护航。
+tags: [Technology,Ecosystem,Authentication]
+---
+
+> 本文为大家带来了 Apache APISIX 即将发布的 Vault 插件以及相关细节。在为服务提供高并发低延迟的卓越性能的同时，为服务的安全保驾护航。
+
+<!--truncate-->
+
+随着微服务架构的兴起，保持服务安全变得比以前更有挑战性。多个后端 server 实例使用单一的静态密钥凭访问数据库 server 会带来巨大的风险。如果发生密钥凭证泄露，整个系统都会受到影响。为了解决密钥凭证泄露所带来的影响，只能撤销这个密钥凭证。而撤销密钥凭证会导致大规模的服务中断。对开发者来说，服务大规模中断是开发者最不想看到事情。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
随着微服务架构的兴起，保持服务安全变得比以前更有挑战性。多个后端 server 实例使用单一的静态密钥凭访问数据库 server 会带来巨大的风险。如果发生密钥凭证泄露，整个系统都会受到影响。为了解决密钥凭证泄露所带来的影响，只能撤销这个密钥凭证。而撤销密钥凭证会导致大规模的服务中断，对开发者来说这是最不想看到事情。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+虽然我们不能提前预知将来会出现哪些安全漏洞，但是我们可以通过配置多个密钥来控制这些安全漏洞的影响范围。为了避免这样的情况，像 HashiCorp Vault （下文简称 Vault）这样密钥凭证解决方案应运而生。
+
+本文演示了如何将 Vault 与 Apache APISIX 的 jwt-auth插件集成，在为服务提供高并发低延迟的卓越性能的同时，为服务的安全保驾护航。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
本文演示了如何将 Vault 与 Apache APISIX 的 `jwt-auth` 插件集成，在为服务提供高并发低延迟的卓越性能的同时，为服务的安全保驾护航。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+
+`jwt-auth` 是一个认证插件，可以附加到任何 APISIX 路由上，在请求被转发到上游 URI 之前执行 JWT 认证。通常情况下，发行者使用私钥或文本密钥来签署 JWT。JWT 的接收者将验证签名，以确保令牌在被发行者签名后没有被改变。整个 JWT 机制的整体完整性取决于签名密钥（或 RSA 密钥对的文本密钥）。这使得未经认证的来源很难猜到签名密钥并试图改变 JWT 中的声明。
+
+因此，在安全的环境中存储这些密钥是非常关键的。如果密钥落入坏人之手，可能会危及整个基础设施的安全。虽然 Apache APISIX 采取了一切手段来遵循标准的 SecOps 实践，但在生产环境中有一个集中的密钥管理解决方案也是一件好事。例如 Vault，有详细的审计日志，定时的密钥轮换，密钥撤销等功能。如果每次在整个基础设施发生密钥轮换时，你都要更新 Apache APISIX 配置，这将是一个相当麻烦的问题。
+
+## 如何集成 Vault 和 Apache APISIX
+
+为了与 Vault 集成，Apache APISIX 需要在 [config.yaml](https://github.com/apache/apisix/blob/master/conf/config.yaml) 文件中加载 Vault 的相关配置信息。
+
+Apache APISIX 与 Vault server [KV secret engine v1](https://www.vaultproject.io/docs/secrets/kv/kv-v1) HTTP [APIs](https://www.vaultproject.io/api/secret/kv/kv-v1) 进行通信。由于大多数企业解决方案倾向于在生产环境中使用 KV Secrets Engine - Version 1，在APISIX-Vault 支持的初始阶段，我们只使用这个版本。在以后的版本中，我们将增加对 K/V - Version 2 的支持。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
Apache APISIX 与 Vault server [KV secret engine v1](https://www.vaultproject.io/docs/secrets/kv/kv-v1) HTTP [APIs](https://www.vaultproject.io/api/secret/kv/kv-v1) 进行通信。由于大多数企业解决方案倾向于在生产环境中使用 KV Secrets Engine - Version 1，在 APISIX-Vault 支持的初始阶段，我们只使用这个版本。在以后的版本中，我们将增加对 K/V - Version 2 的支持。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+
+使用 Vault 而不是 APISIX etcd 后端的主要顾虑是在低信任度环境下的安全问题。因为 Vault 访问令牌是小范围的，可以授予 APISIX server 有限的权限。
+
+### 配置 Vault
+
+本节分享了在 Apache APISIX 生态系统中使用 Vault 的最佳实践。如果你已经有了一个具有必要权限的 Vault 实例在运行，请跳过本节。
+
+#### 启动 Vault server
+
+在这里，你有多种选择，可以自由选择 docker、预编译二进制包或从源代码构建。至于与 Vault server 的通信，你需要一个 Vault CLI 客户端。请运行以下命令启动 server：
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
在这里，你有多种选择，可以自由选择 Docker、预编译二进制包或从源代码构建。至于与 Vault server 的通信，你需要一个 Vault CLI 客户端。请运行以下命令启动 server：
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+Path          Type         Accessor              Description
+----          ----         --------              -----------
+cubbyhole/    cubbyhole    cubbyhole_4eeb394c    per-token private secret storage
+identity/     identity     identity_5ca6201e     identity store
+kv/           kv           kv_92cd6d37           n/a
+secret/       kv           kv_6dd46a53           key/value secret storage
+sys/          system       system_2045ddb1       system endpoints used for control, policy and debugging
+```
+
+#### 第2步：为 Apache APISIX 生成一个 Vault 访问令牌
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
#### 第 2 步：为 Apache APISIX 生成一个 Vault 访问令牌
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+
+本文是关于在 `jwt-auth` 插件中使用 Vault 的观点。因此，对于一个APISIX 消费者 `jack`，`jwt-auth` 插件会在 `<vault.prefix inside config.yaml>/consumer/<consumer.username>/jwt-auth` 中查找（如果启用了 Vault 配置）`secret/s` 到 Vault 键值对 存储。在这种情况下，如果你将 `kv/apisix` 命名空间（Vault 路径）指定为`config.yaml` 内的 `vault.prefix`，用于所有 APISIX 相关数据的检索，我们建议你为路径 `kv/apisix/consumer/` 创建一个策略。最后的星号（*）确保策略允许读取任何具有 `kv/apisix/consumer` 前缀的路径。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
本文是关于在 `jwt-auth` 插件中使用 Vault 的观点。因此，对于一个 APISIX 消费者 `jack`，`jwt-auth` 插件会在 `<vault.prefix inside config.yaml>/consumer/<consumer.username>/jwt-auth` 中查找（如果启用了 Vault 配置）`secret/s` 到 Vault 键值对 存储。在这种情况下，如果你将 `kv/apisix` 命名空间（Vault 路径）指定为`config.yaml` 内的 `vault.prefix`，用于所有 APISIX 相关数据的检索，我们建议你为路径 `kv/apisix/consumer/` 创建一个策略。最后的星号（*）确保策略允许读取任何具有 `kv/apisix/consumer` 前缀的路径。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+vault:
+  host: 'http://0.0.0.0:8200'
+  timeout: 10
+  token: 's.KUWFVhIXgoRuQbbp3j1eMVGa'
+  prefix: 'kv/apisix'
+```
+
+### 创建一个 APISIX Consumer
+
+APISIX 有一个消费者层面的抽象，与认证方案并列。为了启用任何 APISIX 路由的认证，需要一个具有适合该特定类型认证服务的配置的消费者。然后，只有 APISIX 可以通过成功执行消费者配置方面的认证，将请求转发到上游 URI。APISIX 消费者有两个字段：一个是 `username`（必填项），用于识别消费者，另一个是 `plugins`，用于保存消费者所使用的特定插件配置。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
APISIX 有一个消费者层面的抽象，与认证方案并列。为了启用任何 APISIX 路由的认证，需要一个具有适合该特定类型认证服务配置的消费者。之后将通过 APISIX 成功执行消费者配置方面的认证请求转发到上游 URI。APISIX 消费者有两个字段：一个是 `username`（必填项），用于识别消费者；另一个是 `plugins`，用于保存消费者所使用的特定插件配置。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+
+在这里，在这篇文章中，我们将用 `jwt-auth` 插件创建一个消费者。它为各自的路由或服务执行 JWT认证。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
在这里，在这篇文章中，我们将用 `jwt-auth` 插件创建一个消费者。它为各自的路由或服务执行 JWT 认证。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+    "plugins": {
+        "jwt-auth": {
+            "key": "test-key",
+            "vault": {}
+        }
+    }
+}'
+```
+
+在这里，插件在消费者配置中提到的消费者 `jack` 的 Vault 路径（`<vault.prefix from conf.yaml>/consumer/jack/jwt-auth`）中查找密钥 `secret`，并使用它进行后续的签名和 jwt 验证。如果在同一路径中没有找到密钥，该插件会记录错误，并且无法执行 jwt 验证。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
在这里，插件在消费者配置中提到的消费者 `jack` 的 Vault 路径（`<vault.prefix from conf.yaml>/consumer/jack/jwt-auth`）中查找密钥 `secret`，并使用它进行后续的签名和 JWT 验证。如果在同一路径中没有找到密钥，该插件会记录错误，并且无法执行 JWT 验证。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+
+#### 设置一个测试的上游 server
+
+为了测试这个行为，你可以为一个上游创建一个路由（一个简单的 ping 处理程序，返回 pong）。你可以用一个普通的 go HTTP-Server 来设置它。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md:187]: 返回 pong 还是返回 ping

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 11]
```diff
@@ -0,0 +1,379 @@
...
+
+func main() {
+    http.HandleFunc("/secure/ping", ping)
+    http.ListenAndServe(":9999", nil)
+}
+```
+
+#### 创建一个启用了认证的 APISIX 路由
+
+用这个安全的 ping HTTP server和启用了 jwt-auth 认证插件创建一个 APISIX 路由。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
用这个安全的 ping HTTP server 和启用了 `jwt-auth` 认证插件创建一个 APISIX 路由。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 12]
```diff
@@ -0,0 +1,379 @@
...
+        },
+        "type": "roundrobin"
+    },
+    "uri": "/secure/ping"
+}'
+```
+
+#### 从 jwt-auth 插件生成令牌
+
+现在从 APISIX 签署一个 jwt 密文，可以用于并通过向 APISIX server的 `http://localhost:9080/secure/ping` 代理路由发出请求。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
现在从 APISIX 签署一个 JWT 密文，可以用于并通过向 APISIX server 的 `http://localhost:9080/secure/ping` 代理路由发出请求。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 13]
```diff
@@ -0,0 +1,379 @@
...
+Content-Type: text/plain; charset=utf-8
+Transfer-Encoding: chunked
+Connection: keep-alive
+Server: APISIX/2.11.0
+
+
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOiJ0ZXN0LWtleSIsImV4cCI6MTY0MjU3ODY1N30.nkyev1_KUapVgY_QVYETsSApA6gEkDWS8tsHFV1EpD8
+```
+
+在上一步中，如果你看到类似 `failed to sign jwt` 的信息，请确保你有一个私有密钥存储在vault `kv/apisix/consumers/jack/jwt-aut` 路径中。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
在上一步中，如果你看到类似 `failed to sign jwt` 的信息，请确保你有一个私有密钥存储在 vault `kv/apisix/consumers/jack/jwt-aut` 路径中。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 14]
```diff
@@ -0,0 +1,379 @@
...
+Content-Length: 12
+Connection: keep-alive
+Date: Tue, 18 Jan 2022 08:00:04 GMT
+Server: APISIX/2.11.0
+
+
+secure/pong
+```
+
+任何无效的 jwt 请求都会抛出 `HTTP 401 Unauthorized` 的错误。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
任何无效的 JWT 请求都会抛出 `HTTP 401 Unauthorized` 的错误。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 15]
```diff
@@ -0,0 +1,379 @@
...
+                "vault": {}
+            }
+        }
+    }'
+   ```
+
+   The plugin looks up for `public_key` and `private_key` keys inside vault kv path (`<vault.prefix from conf.yaml>/consumer/jim/jwt-auth`) for `jim` mentioned inside plugin vault configuration. If not found, authentication fails.
+   该插件在 Vault 键值对 路径（`<vault.prefix from conf.yaml>/consumer/jim/jwt-auth`）中为插件 Vault 配置中提到的用户 `jim` 查找 `public_key` 和 `private_key`。如果没有找到，认证失败。
+
+   如果你不确定如何将公钥和私钥存储到 Vault 键值对 中，请使用这个命令。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
   如果你不确定如何将公钥和私钥存储到 Vault 键值对中，请使用这个命令。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 16]
```diff
@@ -0,0 +1,379 @@
...
+            }
+        }
+    }'
+   ```
+
+   这个插件使用来自消费者配置的 RSA 公钥，并使用直接从 Vault 获取的私钥。
+
+### 禁用 Vault 插件
+
+现在，要禁用 `jwt-auth` 插件的 Vault 查询，只需从消费者插件配置中删除空的 Vault 对象（本例中是 `jack`）。这将使 jwt 插件在随后对已启用 jwt-auth 配置的 URI 路由的请求中，将查找签名密钥（包括 HS256/HS512 或 RS512 密钥对）纳入插件配置。即使你在 APISIX `config.yaml` 中启用了 Vault 配置，也不会有请求被发送到 Vault server。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
现在，要禁用 `jwt-auth` 插件的 Vault 查询，只需从消费者插件配置中删除空的 Vault 对象（本例中是 `jack`）。这将使 JWT 插件在随后对已启用 `jwt-auth` 配置的 URI 路由的请求中，将查找签名密钥（包括 HS256/HS512 或 RS512 密钥对）纳入插件配置。即使你在 APISIX `config.yaml` 中启用了 Vault 配置，也不会有请求被发送到 Vault server。
```

File: website/blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 17]
```diff
@@ -0,0 +1,373 @@
...
+identity/     identity     identity_5ca6201e     identity store
+kv/           kv           kv_92cd6d37           n/a
+secret/       kv           kv_6dd46a53           key/value secret storage
+sys/          system       system_2045ddb1       system endpoints used for control, policy and debugging
+```
+
+#### Step 2: Generate a Vault Access Token for APISIX
+
+This article is regarding using vault in jwt-auth plugin perspective. So, for an APISIX consumer (if you are unfamiliar with consumers in the APISIX ecosystem, please read the [document about Apache APISIX Consumer](https://apisix.apache.org/docs/apisix/architecture-design/consumer)) with username `jack` the jwt-auth plugin looks up (if enabled with vault configuration) for secret/s at `<vault.prefix inside config.yaml>/consumer/<consumer.username>/jwt-auth` into vault kv storage. In this context, if you are assigning `kv/apisix` namespace (vault path) as `vault.prefix` inside config.yaml for all apisix related data retrieval, we suggest you to create a policy for path `kv/apisix/consumer/*`. The extra asterisk (*) at the end ensure the policy allows read for any path that has a `kv/apisix/consumer` prefix.
```
Comment (SylviaBABY) [inline website/blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
This article is regarding using vault in `jwt-auth` plugin perspective. So, for an APISIX consumer (if you are unfamiliar with consumers in the APISIX ecosystem, please read the [document about Apache APISIX Consumer](https://apisix.apache.org/docs/apisix/architecture-design/consumer)) with username `jack` the `jwt-auth` plugin looks up (if enabled with vault configuration) for secret/s at `<vault.prefix inside config.yaml>/consumer/<consumer.username>/jwt-auth` into vault kv storage. In this context, if you are assigning `kv/apisix` namespace (vault path) as `vault.prefix` inside config.yaml for all apisix related data retrieval, we suggest you to create a policy for path `kv/apisix/consumer/*`. The extra asterisk (*) at the end ensure the policy allows read for any path that has a `kv/apisix/consumer` prefix.
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 18]
```diff
@@ -0,0 +1,379 @@
...
+vault:
+  host: 'http://0.0.0.0:8200'
+  timeout: 10
+  token: 's.KUWFVhIXgoRuQbbp3j1eMVGa'
+  prefix: 'kv/apisix'
+```
+
+### 创建一个 APISIX Consumer
+
+APISIX 有一个消费者层面的抽象，与认证方案并列。为了启用任何 APISIX 路由的认证，需要一个具有适合该特定类型认证服务配置的消费者。之后将通过 APISIX 成功执行消费者配置方面的认证请求转发到上游 URI。APISIX 消费者有两个字段：一个是 `username`（必填项），用于识别消费者；另一个是 `plugins`，用于保存消费者所使用的特定插件配置。
```
Comment (guoqqqi) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]: Are we better off not translating for the `consumer`?

**Review by guoqqqi — DISMISSED** [review]
LGTM!

**Review by juzhiyuan — DISMISSED** [review]

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 19]
```diff
@@ -0,0 +1,379 @@
...
+cubbyhole/    cubbyhole    cubbyhole_4eeb394c    per-token private secret storage
+identity/     identity     identity_5ca6201e     identity store
+kv/           kv           kv_92cd6d37           n/a
+secret/       kv           kv_6dd46a53           key/value secret storage
+sys/          system       system_2045ddb1       system endpoints used for control, policy and debugging
+```
+
+#### 为 Apache APISIX 生成一个 Vault 访问令牌
+
+本文是关于在 `jwt-auth` 插件中使用 Vault 的观点。因此，对于一个 APISIX Consumer `jack`，`jwt-auth` 插件会在 `<vault.prefix inside config.yaml>/consumer/<consumer.username>/jwt-auth` 中查找（如果启用了 Vault 配置）`secret/s` 到 Vault 键值对 存储。在这种情况下，如果你将 `kv/apisix` 命名空间（Vault 路径）指定为`config.yaml` 内的 `vault.prefix`，用于所有 APISIX 相关数据的检索，我们建议你为路径 `kv/apisix/consumer/` 创建一个策略。最后的星号（*）确保策略允许读取任何具有 `kv/apisix/consumer` 前缀的路径。
```
Comment (bzp2010) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md]: > 本文是关于在 `jwt-auth` 插件中使用 Vault 的观点。

本文是关于如何在 `jwt-auth` 插件中使用 Vault 的介绍。

File: website/blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 20]
```diff
@@ -0,0 +1,373 @@
...
+  - name: "Yilin Zeng"
+    title: "Technical Writer"
+    url: "https://github.com/yzeng25"
+    image_url: "https://avatars.githubusercontent.com/u/36651058?v=4"
+keywords: 
+- Apache APISIX
+- HashiCorp
+- Vault
+- jwt-auth
+- Ecosystem
```
Comment (bisakhmondal) [inline website/blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
- Apache APISIX
- HashiCorp
- Vault
- jwt-auth
- Authentication
```

File: website/blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 21]
```diff
@@ -0,0 +1,373 @@
...
+description: This article brings you the upcoming release of the Vault-Apache APISIX integration and related details.
+tags: [Technology,Ecosystem,Authentication]
```
Comment (bisakhmondal) [inline website/blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
tags: [Technology,Authentication]
```
`Ecosystem` doesn't go with it very well. WDYT?

File: website/blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 22]
```diff
@@ -0,0 +1,373 @@
...
+        }
+    }
+}'
+```
+
+## Summary
+
+This article brings you the upcoming release of the Vault-Apache APISIX integration and related details.
+
+Feel free to start a discussion in [GitHub Discussions](https://github.com/apache/apisix/discussions) or communicate via the [mailing list](https://apisix.apache.org/zh/docs/general/subscribe-guide).
```
Comment (bisakhmondal) [inline website/blog/2022/01/21/apisix-hashicorp-vault-integration.md]:
```suggestion
Feel free to start a discussion in [GitHub Discussions](https://github.com/apache/apisix/discussions) or communicate via the [mailing list](https://apisix.apache.org/docs/general/subscribe-guide).
```

Btw, just wondering if it is possible to add a ref to my blog? Hehe😅

**Review by bisakhmondal — DISMISSED** [review]
Great. Thanks @yzeng25 for doing this : )

File: website/blog/2022/01/21/apisix-hashicorp-vault-integration.md (hunk 22 above)
Comment (yzeng25) [reply website/blog/2022/01/21/apisix-hashicorp-vault-integration.md]: sure

**Review by takumi913 — APPROVED** [review]

**Comment by takumi913** [comment]
LGTM

**Review by SylviaBABY — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Review by okaybase — APPROVED** [review]

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 23]
```diff
@@ -0,0 +1,383 @@
...
+            "jwt-auth": {
+                "key": "rsa-keypair",
+                "algorithm": "RS256",
+                "vault": {}
+            }
+        }
+    }'
+   ```
+
+   The plugin looks up for `public_key` and `private_key` keys inside vault kv path (`<vault.prefix from conf.yaml>/consumer/jim/jwt-auth`) for `jim` mentioned inside plugin vault configuration. If not found, authentication fails.
```
Comment (guoqqqi) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md:326]:
```suggestion

```
Excess?

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md [hunk 24]
```diff
@@ -0,0 +1,383 @@
...
+   该插件在 Vault 键值对 路径（`<vault.prefix from conf.yaml>/consumer/jim/jwt-auth`）中为插件 Vault 配置中提到的用户 `jim` 查找 `public_key` 和 `private_key`。如果没有找到，认证失败。
```
Comment (guoqqqi) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md:327]:
```suggestion
   该插件在 Vault 键值对路径（`<vault.prefix from conf.yaml>/consumer/jim/jwt-auth`）中为插件 Vault 配置中提到的用户 `jim` 查找 `public_key` 和 `private_key`。如果没有找到，认证失败。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md (hunk 23 above)
Comment (yzeng25) [reply website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md:326]: solved in https://github.com/apache/apisix-website/pull/857

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md (hunk 24 above)
Comment (yzeng25) [reply website/i18n/zh/docusaurus-plugin-content-blog/2022/01/21/apisix-hashicorp-vault-integration.md:327]: solved in https://github.com/apache/apisix-website/pull/857

**Pull request merged**